src/lib/Types/Primitives.hs

```diff
@@ -151,7 +150,7 @@ traversePrimOp = inline traverse
 {-# INLINABLE traversePrimOp #-}
 
 data PrimHof e =
-        For ForAnn e e        -- Ix type, body lambda
+        For ForAnn e e e        -- index type, ix dict, body lambda
```

Comment:
We could probably query the index type from the dict, couldn't we? 🤔

Reply:
Good point. Just pushed a change to do it that way instead. At the very least it should reduce IR sizes.